src/main/groovy/com/bmuschko/gradle/docker/tasks/image/Dockerfile.groovy

```diff
@@ -373,6 +373,30 @@ class Dockerfile extends DefaultTask {
         }
     }
 
+    static abstract class MapInstruction implements Instruction {
+        final Object command
+
+        MapInstruction(Map<String, String> command) {
+            this.command = command
+        }
+
+        MapInstruction(Closure command) {
+            this.command = command
+        }
+
+        @Override
+        String build() {
+            if (command instanceof Map<String, String>) {
+                String args = ""
+                ((Map<String, String>) command).each { key, value -> args += " \"$key\"=\"$value\"" }
+                keyword + args
```

Comment:
This was mildly concerning to me.  Originally I was inserting a space between the keyword (e.g. "LABEL") and the arguments.  However, the unit tests were failing due to an extra space character... because something else is apparently already adding a space on the end of the keyword.  I couldn't quite figure that out (Groovy isn't my native tongue), so I'm relying on it here and assuming that the space will be present.